server/routes/createGoal/CreateGoalController.ts

```diff
@@ -24,13 +24,18 @@ export default class CreateGoalController {
     })
     const processedData: NewGoal = this.processGoalData(req.body)
     const planUuid = req.services.sessionService.getPlanUUID()
-    const { uuid } = await this.goalService.saveGoal(processedData, planUuid)
-    req.services.formStorageService.saveFormData('currentGoal', {
-      processed: null,
-      raw: { uuid },
-    })
-    const redirectUrl: string = req.body.action === 'addStep' ? URLs.CREATE_STEP : `${URLs.PLAN_SUMMARY}?status=success`
-    return res.redirect(redirectUrl)
+    try {
+      const { uuid } = await this.goalService.saveGoal(processedData, planUuid)
+      req.services.formStorageService.saveFormData('currentGoal', {
+        processed: null,
+        raw: { uuid },
+      })
+      const redirectUrl: string =
+        req.body.action === 'addStep' ? URLs.CREATE_STEP : `${URLs.PLAN_SUMMARY}?status=success`
+      return res.redirect(redirectUrl)
+    } catch (e) {
```

Comment:
what's the error this catches? In fact, is it an error or something else?

Reply:
A `SanitisedError`. The call to `saveGoal()` will fail (as the API will reject it as being too ling for the column definition) if a goal title of greater than 128 chars is sent. The app falls over if this happens, as express 4 will not handle the async error.

Reply:
awesome.